docs/bundles.rst

```diff
@@ -0,0 +1,137 @@
+  url_2 = blockstore_service.url_for_path(path_of_file_2)
+
+For directories or files in the same Bundle as the XBlock, the URI must be an absolute-path reference
+which consists of a single slash character followed by the full path of the directory or file from the root
+of the Bundle.
+
+.. code-block::
+
+  /bundle.json
+  /mcqs/mcq1.olx
+  /b/c/icon.png
+  /d/README
+  /e/f/
+
+For files in other Bundles, the URI must be a network-path reference which begins with two slash characters followed
+by the alias of the relevant Bundle version as specified in ``bundle.json`` followed by the absolute-path reference
+of the directory or file from the root of its Bundle.
+
+.. code-block::
+
+  //problems/description.olx
+  //problems/mcqs/mcq1.olx
+  //problems/images/
+  //problems/resources/intro.pdf
+  //videos_lectures/vid/lecture1.olx
+
+If relative-path references are used in files, XBlocks will be responsible for prepending the URI of the relevant
+base directory when necessary.
+
+bundle.json
+-------------------------
+The bundle.json file contains metadata about the Bundle. Authoring clients must add and update this themselves.
+It can contain the following fields:
+
+schema
+~~~~~~~~~~~~~~~~~~~~~~
+
+The **schema** field is required and specifies the document format version.
+
+.. code-block::
+
+  "schema": 0.1
+
+type
+~~~~~~~~~~~~~~~~~~~~~~
+
+The **type** field is required and should specify the type of content in the Bundle. Its value must be one of:
+
+- olx/collection
+- olx/course
+- olx/sequence
+- olx/unit
+- static
+
+.. code-block::
+
+    "type": "olx/collection"
+
+components
+~~~~~~~~~~~~~~~~~~~~~~
+
+The optional **components** field is an array of absolute paths to OLX files for XBlocks in the Bundle the author wants
+available for use independently. These may be shown in previews of the Bundle so that authors can select them
+for use.
+
+.. code-block::
+
+    "components": [
+      "/description.olx",
+      "/mcqs/mcq1.olx",
+      "/mcqs/mcq2.olx",
+      "/mcqs/mcq3.olx"
+    ]
+
+It is valid to not specify **components**. This may be useful in cases like having a shared Bundle with
+branding assets only.
+
+assets
+~~~~~~~~~~~~~~~~~~~~~~
+
+The optional **assets** field is an array of absolute paths to directories and files in the Bundle which
+should be internet accessible from client applications via URLs.
+
+.. code-block::
+
+  "assets": {
+    "/images",
+    "/resources/intro.pdf"
+  }
+
+dependencies
+~~~~~~~~~~~~~~~~~~~~~~
+
+The optional **dependencies** field is a map of aliases to bundle version info objects. The later must have
+two fields: **bundle_uuid** and **version_num**.
+
+If a file from another Bundle is to be referenced, a version of that Bundle must be specified in this field.
+URIs to directories or files in other Bundles use the alias as the authority.
+
+.. code-block::
+
+  "dependencies": {
+    "problems": {
+      "bundle_uuid": "21d45e735e134c41ae3b24fde26d4369",
```

Comment:
UUIDs are normally written with hyphens - can we add them in here, and should we specify whether or not hyphens are allowed?

Reply:
Ah, yes! The canonical representation uses dashes according to section 3 of https://tools.ietf.org/html/rfc4122#section-4.1. I think I just copied a block id from an olx file here. Will fix this and add the format requirement.